apis/kueue/v1alpha1/cohort_types.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kueuebeta "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+)
+
+// CohortSpec defines the desired state of Cohort
+type CohortSpec struct {
+	// Parent references the name of the Cohort's parent, if
+	// any. It satisfies one of three cases:
+	// 1) Unset. This Cohort is the root of its Cohort tree.
+	// 2) References a non-existent Cohort. We use default Cohort (no borrowing/lending limits).
+	// 3) References an existent Cohort.
+	//
+	//+kubebuilder:validation:MaxLength=253
+	//+kubebuilder:validation:Pattern="^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
+	//
+	Parent string `json:"parent,omitempty"`
+
+	// ResourceGroups describes groupings of Resources and
+	// Flavors.  Each ResourceGroup defines a list of Resources
+	// and a list of Flavors which provide quotas for these
+	// Resources. Each Resource and each Flavor may only form part
+	// of one ResourceGroup.  There may be up to 16 ResourceGroups
+	// within a Cohort.
+	//
+	// BorrowingLimit limits how much members of this Cohort
+	// subtree can borrow from the parent subtree. This limit must
+	// only be set when the Cohort has a parent.
```

Comment:
What happens otherwise? We catch it at validation phase, invalidate the cohort or let it be?

Reply:
This will be validated by the webhook, and we will reject the create/update. Updated the documentation.